sonar-pmd-plugin/src/main/java/org/sonar/plugins/pmd/PmdViolationRecorder.java

```diff
@@ -72,7 +73,7 @@ public void saveViolation(RuleViolation pmdViolation, SensorContext context) {
     }
 
     private InputFile findResourceFor(RuleViolation violation) {
-        final URI uri = URI.create(violation.getFilename());
+        final URI uri = Paths.get(violation.getFilename()).toUri();
```

Comment:
fix throw IllegalArgumentException: "Missing scheme" for URI.create("path/to/file").schema === null.